### DISCUSSION TITLE
DatePickerInput controlled date

### DISCUSSION THREAD

**Discussion by hensansi** [mantinedev/mantine#4158] [Q&A] [answered]
I'm trying to do the following with the datepicker:
- set as default startDate as today and endDate in 1 year
- allow user to changes dates by displaying startDate -> user can pick anything
- jump to startDate + 1 year -> allow user to pick anything close to it, it could be a couple of days or a couple of weeks in the future.

The current problem with the datepicker is that the controlled way to focus the date, the prop `date` disables the next navigation, or locks the user to the current month and doesn't allow to jump to a previous or next month. Is there a way to do what I need?

https://codesandbox.io/s/controlled-range-datepicker-9ke4j2?file=/src/App.tsx

**Comment by rtivital** [comment, marked as answer]
There is excatly this example in documentation – https://mantine.dev/dates/date-picker/#controlled-date

**Reply by hensansi** [reply]
Thanks I improved the example a bit for my use case, as in it should jump to startDate the second time around in the case of the Picker,
https://codesandbox.io/s/controlled-range-datepicker-9ke4j2?file=/src/App.tsx


I was missing the `onDateChange={setDate}` on my code
